packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -81,6 +81,9 @@ typedef ActionListenerCallback = void Function(Action<Intent> action);
 ///  * [ActionDispatcher], a class that takes an [Action] and invokes it, passing
 ///    a given [Intent].
 abstract class Action<T extends Intent> with Diagnosticable {
+  Action();
+  factory Action.overrideableAction(Action<T> defaultAction, BuildContext lookupContext) => defaultAction.makeOverridableAction(lookupContext);
```

Comment:
@HansMuller this is the factory constructor version. All I had to add is these 2 lines (and make `makeOverridableAction` private). so it's
```dart
Action.overrideableAction(
  defaultAction: CallbackAction<MyIntent>(
    onInvoke: (MyIntent intent) => print(intent),
  ),
  lookupContext: context,
);
```
versus
```dart
CallbackAction<MyIntent>(
  onInvoke: (MyIntent intent) => print(intent),
).makeOverridableAction(context)
```

Reply:
This looks good however I don't think "find the closest enabled action" is the right semantics for the override. I don't think the overriding action shouldn't lose the ability to disable the action.

Reply:
To disable an action you can map the intent to `DoNothingAction`?

Reply:
You could, although the semantics of that are different, since [isEnabled](https://api.flutter.dev/flutter/widgets/Action/isEnabled.html) is wired to the action's listeners. The most straightforward interpretation of what we're supporting here  is overriding an _action_ without any qualification. For example, I don't think it would be any more intuitive to only search for actions whose value was not `DoNothingAction`.

Reply:
BTW - I think `Action.overrideableAction()` is redundant, could be `Action.overrideable()`